src/Runtime.Base/src/System/Runtime/CachedInterfaceDispatch.cs

```diff
@@ -17,8 +17,7 @@ internal unsafe static class CachedInterfaceDispatch
         unsafe private static IntPtr RhpCidResolve(IntPtr callerTransitionBlockParam, IntPtr pCell)
         {
             IntPtr locationOfThisPointer = callerTransitionBlockParam + TransitionBlock.GetThisOffset();
-            IntPtr thisPointerAsIntPtr = *(IntPtr*)locationOfThisPointer;
-            object pObject = RuntimeAugments.ConvertIntPtrToObjectReference(thisPointerAsIntPtr);
+            object pObject = Unsafe.As<IntPtr, Object>(ref *(IntPtr*)locationOfThisPointer);
```

Comment:
I received the necessary signoff from the codegen team and the larger cleanup is now en route to the nmirror branch.